packages/pages/src/dev/server/middleware/serverRenderSlugRoute.ts

```diff
@@ -19,32 +21,45 @@ type Props = {
 
 export const serverRenderSlugRoute =
   ({ vite, dynamicGenerateData, projectStructure }: Props): RequestHandler =>
-  async (req, res, next) => {
+  async (req, res, next): Promise<void> => {
     try {
       const url = new URL("http://" + req.headers.host + req.originalUrl);
-      // Right now we're assuming this is the slug and not a static URL
-      const slug = url.pathname.substring(1);
       const locale = req.query.locale?.toString() ?? "en";
+      const slug = url.pathname.substring(1);
+
       const templateFilepaths =
-        getTemplateFilePathsFromProjectStructure(projectStructure);
+        getTemplateFilepathsFromProjectStructure(projectStructure);
+      const matchingStaticTemplate: TemplateModuleInternal<any, any> | null =
+        await findMatchingStaticTemplate(vite, slug, templateFilepaths);
+      if (matchingStaticTemplate) {
+        sendStaticPage(res, vite, matchingStaticTemplate, locale, url.pathname);
+        return;
+      }
+
       const document = await getDocument(
         dynamicGenerateData,
         slug,
         locale,
         projectStructure
       );
+      if (!document) {
+        send404(res, `Cannot find document corresponding to slug: ${slug}`);
+        return;
+      }
+
       const feature = document.__.name;
       const entityId = document.id;
-      const templateModuleInternal = await featureNameToTemplateModuleInternal(
+      const templateModuleInternal = await findTemplateModuleInternal(
         vite,
-        feature,
+        (t) => feature === t.config.name,
```

Comment:
I think this is fine, but double check that the templateModuleInternal's config name is properly set to the either the config.name in the template OR the name of the file if it's not present. And that that same thing is being set in the document.__.name.